### PR TITLE
provide RTL mode for dxDropDownBox/dxDropDownButton (T845269)

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.editors/dropDownBox.tests.js [hunk 1]
```diff
@@ -399,6 +399,26 @@ QUnit.module("popup options", moduleConfig, () => {
...
+            dropDownOptions: {
+                width: 200,
+                "position.collision": "none"
+            },
+        });
+
+        const dropDownButtonElementRect = this.$element.get(0).getBoundingClientRect();
+
+        let popupContentElementRect = instance.content().getBoundingClientRect();
+        assert.equal(popupContentElementRect.left, dropDownButtonElementRect.left, "popup position is right, rtlEnabled = false");
```
Comment (alexander-kotov-dx) [inline testing/tests/DevExpress.ui.widgets.editors/dropDownBox.tests.js]: strictEqual ?

Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets.editors/dropDownBox.tests.js]: fixed

File: testing/tests/DevExpress.ui.widgets.editors/dropDownBox.tests.js [hunk 2]
```diff
@@ -399,6 +399,26 @@ QUnit.module("popup options", moduleConfig, () => {
         assert.equal($popupContent.outerWidth(), 200, "popup width customization has been cancelled");
     });
 
+    QUnit.test("popup should be positioned right if rtlEnabled is true", function(assert) {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/dropDownBox.tests.js]:
```suggestion
    QUnit.test("popup should be positioned correctly if rtlEnabled is true", function(assert) {
```

File: testing/tests/DevExpress.ui.widgets.editors/dropDownBox.tests.js (hunk 1 above)
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/dropDownBox.tests.js]:
```suggestion
        assert.strictEqual(popupContentElementRect.left, dropDownButtonElementRect.left, "popup position is right, rtlEnabled = false");
```

File: testing/tests/DevExpress.ui.widgets.editors/dropDownBox.tests.js [hunk 3]
```diff
@@ -399,6 +399,26 @@ QUnit.module("popup options", moduleConfig, () => {
...
+        });
+
+        const dropDownButtonElementRect = this.$element.get(0).getBoundingClientRect();
+
+        let popupContentElementRect = instance.content().getBoundingClientRect();
+        assert.equal(popupContentElementRect.left, dropDownButtonElementRect.left, "popup position is right, rtlEnabled = false");
+
+        instance.option("rtlEnabled", true);
+        popupContentElementRect = instance.content().getBoundingClientRect();
+        assert.equal(popupContentElementRect.right, dropDownButtonElementRect.right, "popup position is right, rtlEnabled = true");
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/dropDownBox.tests.js]:
```suggestion
        assert.strictEqual(popupContentElementRect.right, dropDownButtonElementRect.right, "popup position is right, rtlEnabled = true");
```

File: testing/tests/DevExpress.ui.widgets/overlay.tests.js [hunk 4]
```diff
@@ -350,6 +350,17 @@ testModule("option", moduleConfig, () => {
         assert.strictEqual(contentRenderSpy.callCount, 2, "must invalidate content when RTL changed");
     });
 
+    QUnit.test("overlay should not change visibility after rtlEnabled option change", function(assert) {
+        const overlay = $("#overlay").dxOverlay({
+            visible: true
+        }).dxOverlay("instance");
+
+        assert.ok(overlay.option("visible"), "overlay is visible");
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/overlay.tests.js]: Do we need this check?

**Review by San4es — APPROVED** [review]
LGTM

**Review by alexander-kotov-dx — APPROVED** [review]

**Pull request merged**